ihp-hsx/IHP/HSX/QQ.hs

```diff
@@ -27,20 +27,40 @@ import Data.List (foldl')
 import IHP.HSX.Attribute
 import qualified Text.Blaze.Html5.Attributes as Attributes
 import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Set as Set
 
 hsx :: QuasiQuoter
 hsx = QuasiQuoter {
-        quoteExp = quoteHsxExpression,
+        quoteExp = quoteHsxExpression (HsxSettings True Set.empty Set.empty),
         quotePat = error "quotePat: not defined",
         quoteDec = error "quoteDec: not defined",
         quoteType = error "quoteType: not defined"
     }
 
-quoteHsxExpression :: String -> TH.ExpQ
-quoteHsxExpression code = do
+uncheckedHsx :: QuasiQuoter
+uncheckedHsx = QuasiQuoter {
+        quoteExp = quoteHsxExpression (HsxSettings False Set.empty Set.empty),
+        quotePat = error "quotePat: not defined",
+        quoteDec = error "quoteDec: not defined",
+        quoteType = error "quoteType: not defined"
+    }
+
+newtype AdditionalTags = AdditionalTags [Text]
+newtype AdditionalAttributes = AdditionalAttributes [Text]
+
+customHsx :: AdditionalTags -> AdditionalAttributes -> QuasiQuoter
+customHsx (AdditionalTags additionalTagNames) (AdditionalAttributes additionalAttributeNames) = QuasiQuoter {
```

Comment:
How about `customHsx :: HsxSettings -> QuasiQuoter`? This way it's more flexible if we add more options in the future

Reply:
no problem